src/Utilities/cachingoptimizer.jl

```diff
@@ -901,51 +916,33 @@ function MOI.get(
     attr::Union{MOI.AbstractVariableAttribute,MOI.AbstractConstraintAttribute},
     index::MOI.Index,
 )
-    if MOI.is_set_by_optimize(attr)
-        if state(model) == NO_OPTIMIZER
-            error(
-                "Cannot query $(attr) from caching optimizer because no " *
-                "optimizer is attached.",
-            )
-        end
-        return map_indices(
```

Comment:
@blegat the problematic error because we haven't checked for `state(model) == EMPTY_OPTIMIZER` here. 

and there's a complication: a call like `index_map, false = optimize!(dest, src)` leaves the model in `EMPTY_OPTIMIZER`, even though `model_to_optimizer_map` is populated.

Reply:
It seems to me `needs_optimizer_map::Bool = false` should the trick, I can't of a corner case for which your idea won't work